src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -487,13 +494,28 @@ private async Task ResetStatesAsync()
                         {
                             throw ExceptionUtilities.Unreachable;
                         }
```

Comment:
this catch seems to broadly cover a lot of code. can it be narrowed to only a specific region of code we're trying to protect?

Reply:
it is fatal watson. catch is here since we wanted to have fatal watson without await magic thing wrapped actual exception.